lib/logstash/filters/jdbc/read_write_database.rb

```diff
@@ -0,0 +1,86 @@
+require_relative "basic_database"
+
+module LogStash module Filters module Jdbc
+  class ReadWriteDatabase < BasicDatabase
+    def repopulate_all(loaders)
+      case loaders.size
```

Comment:
Is the `case` statement needed here? Doesn't the `else` satisfy the other cases too?

Reply:
I am hoping JRuby will optimise for the size == 1 or 2 cases because the loaders will never change while a pipeline is running. I might be completely wrong about this and it won't optimise this. OTOH IMO 98% of the time will be size 1 or 2 so it avoids the block calls.